File: frontend/src/shared/models/database/flow-point-dto.ts
import { AreaSizingModeEnum } from "@/shared/enums/backend/area/area-sizing-mode-enum";

export class FlowPointDto {
  id: number = 0;
  name: string = "";

  // Frame this point lives in. Null = absolute screen, which needs rebinding after an import.
  flowAreaId?: number | null;
  offsetMode: AreaSizingModeEnum = AreaSizingModeEnum.ABSOLUTE_PX;

  locationX: number = 0;
  locationY: number = 0;

  ratioX: number = 0;
  ratioY: number = 0;

  flowId: number = 0;

  // Read only, set by the backend.
  flowStepsCount: number = 0;
  flowAreaName: string = "";

  constructor(data: Partial<FlowPointDto> = {}) {
    Object.assign(this, {
      ...data,
    });
  }
}

File: backend/Core/Models/Dtos/Database/FlowPointDto.cs
using Core.Enums;

namespace Core.Models.Dtos
{
    public class FlowPointDto
    {
        /// <summary>0 for a new row. Negative for a new row that references another new row.</summary>
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public int? FlowAreaId { get; set; }
        public AreaSizingModeEnum OffsetMode { get; set; }

        public int LocationX { get; set; }
        public int LocationY { get; set; }

        public float RatioX { get; set; }
        public float RatioY { get; set; }

        public int FlowId { get; set; }

        // Read only, projected by the query.
        public int FlowStepsCount { get; set; }
        public string FlowAreaName { get; set; } = string.Empty;
    }
}
